docs/en/sql-reference/aggregate-functions/reference/count.md

```diff
@@ -29,7 +29,7 @@ In both cases the type of the returned value is [UInt64](../../../sql-reference/
 
 ClickHouse supports the `COUNT(DISTINCT ...)` syntax. The behavior of this construction depends on the [count_distinct_implementation](../../../operations/settings/settings.md#settings-count_distinct_implementation) setting. It defines which of the [uniq\*](../../../sql-reference/aggregate-functions/reference/uniq.md#agg_function-uniq) functions is used to perform the operation. The default is the [uniqExact](../../../sql-reference/aggregate-functions/reference/uniqexact.md#agg_function-uniqexact) function.
 
-The `SELECT count() FROM table` query is not optimized, because the number of entries in the table is not stored separately. It chooses a small column from the table and counts the number of values in it.
+The `SELECT count() FROM table` query can be optimized by enabling the [optimize_functions_to_subcolumns](../../../operations/settings/settings.md#optimize-functions-to-subcolumns) setting. With `optimize_functions_to_subcolumns = 1` the function reads only [null](../../../sql-reference/data-types/nullable.md#finding-null) subcolumn instead of reading and processing the whole table data.
```

Comment:
It's misleading. Only the query `SELECT count(nullable_column) FROM table` will be optimized. Need to mention it in separate sentence.

However, sentence in current version is also outdated, because there exists setting `optimize_trivial_count_query`, but it's not related to this task.

Reply:
BTW We can create a task to document this setting, but I cannot find a PR in which the setting was introduced.

Reply:
It was introduced here #7510.